<a href="https://colab.research.google.com/github/mongikourchid/Prediction-of-Product-Sales/blob/main/Project_1_Part_2_(Core).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>


*   Author:Kourchid mongi





# ***Project 1 - Parts 2(Core)***

**Imports**

In [2]:
import pandas as pd
import numpy as np

# **Loading Data**

Use Pandas to read the sales prediction data set into Google Colab.
Use df.info() and df.head() methods to preview the first rows of your data and a summary of your DataFrame's columns.

In [29]:
df = pd.read_csv('/content/drive/MyDrive/Week 2/sales_predictions_2023.csv')
df.head()

Item_Identifier  Item_Weight Item_Fat_Content  Item_Visibility  \
0           FDA15         9.30          Low Fat         0.016047   
1           DRC01         5.92          Regular         0.019278   
2           FDN15        17.50          Low Fat         0.016760   
3           FDX07        19.20          Regular         0.000000   
4           NCD19         8.93          Low Fat         0.000000   

               Item_Type  Item_MRP Outlet_Identifier  \
0                  Dairy  249.8092            OUT049   
1            Soft Drinks   48.2692            OUT018   
2                   Meat  141.6180            OUT049   
3  Fruits and Vegetables  182.0950            OUT010   
4              Household   53.8614            OUT013   

   Outlet_Establishment_Year Outlet_Size Outlet_Location_Type  \
0                       1999      Medium               Tier 1   
1                       2009      Medium               Tier 3   
2                       1999      Medium               Tier 1   
3                       1998         NaN               Tier 3   
4                       1987        High               Tier 3   

         Outlet_Type  Item_Outlet_Sales  
0  Supermarket Type1          3735.1380  
1  Supermarket Type2           443.4228  
2  Supermarket Type1          2097.2700  
3      Grocery Store           732.3800  
4  Supermarket Type1           994.7052

In [15]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


# Clean Data

1) How many rows and columns?

In [30]:
df.shape

(8523, 12)

 - There are 8523 rows and 12 columns

2) What are the datatypes of each variable?

In [13]:
df.dtypes

Item_Identifier               object
Item_Weight                  float64
Item_Fat_Content              object
Item_Visibility              float64
Item_Type                     object
Item_MRP                     float64
Outlet_Identifier             object
Outlet_Establishment_Year      int64
Outlet_Size                   object
Outlet_Location_Type          object
Outlet_Type                   object
Item_Outlet_Sales            float64
dtype: object

3) Are there duplicates? If so, drop any duplicates.

* the Item_Outlet_Sales  column is showing as an object type column and should be either a float64 or an int64 type column
*the Outlet_Size column contains NAN value
*All other data types are as they should be.

4) Identifiez les valeurs manquantes

In [31]:
# This will give us the total number of null values (True) for each column
null_sums = df.isna().sum()
null_sums

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

In [32]:
# Check the datatype
type(null_sums)



pandas.core.series.Series

In [33]:
# Get the % of null values
null_percent = null_sums/len(df) * 100
null_percent



Item_Identifier               0.000000
Item_Weight                  17.165317
Item_Fat_Content              0.000000
Item_Visibility               0.000000
Item_Type                     0.000000
Item_MRP                      0.000000
Outlet_Identifier             0.000000
Outlet_Establishment_Year     0.000000
Outlet_Size                  28.276428
Outlet_Location_Type          0.000000
Outlet_Type                   0.000000
Item_Outlet_Sales             0.000000
dtype: float64

In [34]:
# Dropping Null values from Item_Weight
df = df.dropna(subset=["Item_Weight"])

In [35]:
# Confirming no more null Item_Weight
df['Item_Weight'].isna().sum()

0

In [36]:
# Dropping Null values from Item_Weight
df = df.dropna(subset=["Outlet_Size"])

In [37]:
# Confirming no more null Item_Weight
df['Outlet_Size'].isna().sum()

0

In [40]:
# Saving list of column names for object cols
cat_cols = df.select_dtypes('object').columns
cat_cols

Index(['Item_Identifier', 'Item_Fat_Content', 'Item_Type', 'Outlet_Identifier',
       'Outlet_Size', 'Outlet_Location_Type', 'Outlet_Type'],
      dtype='object')

In [41]:
# Saving list of column names for numeric cols
num_cols = df.select_dtypes('number').columns
num_cols

Index(['Item_Weight', 'Item_Visibility', 'Item_MRP',
       'Outlet_Establishment_Year', 'Item_Outlet_Sales'],
      dtype='object')

In [42]:
# Checking which categorical columns had missing values.
df[cat_cols].isna().sum()

Item_Identifier         0
Item_Fat_Content        0
Item_Type               0
Outlet_Identifier       0
Outlet_Size             0
Outlet_Location_Type    0
Outlet_Type             0
dtype: int64

8) For any numerical columns, obtain the summary statistics of each (min, max, mean).**texte en gras**

In [45]:
# Obtention des statistiques récapitulatives pour les colonnes numériques
summary_stats = df.describe()
print(summary_stats)

       Item_Weight  Item_Visibility     Item_MRP  Outlet_Establishment_Year  \
count  4650.000000      4650.000000  4650.000000                4650.000000   
mean     12.898675         0.060700   141.716328                1999.190538   
std       4.670973         0.044607    62.420534                   7.388800   
min       4.555000         0.000000    31.490000                1987.000000   
25%       8.770000         0.025968    94.409400                1997.000000   
50%      12.650000         0.049655   142.979900                1999.000000   
75%      17.000000         0.088736   186.614150                2004.000000   
max      21.350000         0.188323   266.888400                2009.000000   

       Item_Outlet_Sales  
count        4650.000000  
mean         2272.037489  
std          1497.964740  
min            69.243200  
25%          1125.202000  
50%          1939.808300  
75%          3111.616300  
max         10256.649000  
